src/shogun/machine/GLM.h

```diff
@@ -0,0 +1,293 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ * 
+ * Both the documentation and the code is heavily inspired by pyGLMnet.: https://github.com/glm-tools/pyglmnet/
+ *
+ * Author: Tej Sukhatme
+ */
+
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/machine/Machine.h>
+#include <shogun/machine/LinearMachine.h>
+#include <shogun/machine/IterativeMachine.h>
+#include <shogun/mathematics/RandomMixin.h>
+#include <shogun/distributions/Distribution.h>
+#include <shogun/optimization/GradientDescendUpdater.h>
+#include <shogun/optimization/ConstLearningRate.h>
+#include <shogun/optimization/ElasticNetPenalty.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/optimization/FirstOrderCostFunction.h>
+
+namespace shogun
+{
+    enum GLM_DISTRIBUTION
+	{
+        //TODO GAUSSIAN,
+        //TODO BINOMIAL,
+        //TODO GAMMA,
+        //TODO SOFTPLUS,
+        //TODO PROBIT,
+        POISSON
+	};
+
+class DotFeatures;
+class Features;
+class RegressionLabels;
+class GLMCostFunction;
+
+/** @brief Class for estimating regularized generalized linear models (GLM).
+ *   The regularized GLM minimizes the penalized negative log likelihood:
+ *
+ *  This uses Elastic-net penalty which defaults to the ridge penalty when
+ *  alpha = 0 and defaults to the lasso penalty when alpha = 1.
+ *
+ * */
+class GLM : public RandomMixin<IterativeMachine<LinearMachine>>
+{
+ 	public:
+		friend class GLMCostFunction;
+		
+		/** default constructor */
+		GLM();
+
+ 		GLM(GLM_DISTRIBUTION distribution, float64_t alpha, float64_t lambda, float64_t learning_rate, int32_t max_iterations, float64_t tolerance, float64_t eta);
```

Comment:
nit: line too long

Reply:
I have fixed this, however, is the 80 character limit rule supposed to be followed strictly? https://github.com/shogun-toolbox/shogun/wiki/Code-style#other-general-rules
As many other lines of mine are crossing the 80 character limit.

Reply:
In order to fix the code style all at once, you can use the script called `check_format.sh` inside the `scripts` directory. You can call it this way
```bash
./check_format.sh feature/implementGLM develop
```
It will give you all the info needed to fix the style automatically. You need to have installed `clang-format` to do use it.

Reply:
it is also very simple to do this by hand btw (at least for the 80 char limit)